# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 50475885250c
# Full Hash: 50475885250c5a0411ae055f765bb7506dc41337
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 12 - Make BrowserHost implement nsIRemoteTab by delegating to nsIRemoteTab. r=nika
#   
#   This commit implements nsIRemoteTab in BrowserHost by delegating to nsIRemoteTab. In a
#   future commit, these methods will be implemented by BrowserHost.
#   
# ==============================================================================

diff -r 4cf9a1ce9e92 -r 50475885250c dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Mon May 06 21:29:48 2019 -0500
+++ b/dom/base/nsFrameLoader.cpp	Wed May 08 11:45:23 2019 -0500
@@ -3275,7 +3275,13 @@
 }
 
 already_AddRefed<nsIRemoteTab> nsFrameLoader::GetRemoteTab() {
-  return do_AddRef(GetBrowserParent());
+  if (!mRemoteBrowser) {
+    return nullptr;
+  }
+  if (auto* browserHost = mRemoteBrowser->AsBrowserHost()) {
+    return do_AddRef(browserHost);
+  }
+  return nullptr;
 }
 
 already_AddRefed<nsILoadContext> nsFrameLoader::LoadContext() {